### PR TITLE
Test auto wrapper

### PR DESCRIPTION
**Pull request by gpzlx1**
I implement `gs.jit.compile`，which supports auto-wrapper for user's code.
Example
```python
import gs
import torch


@gs.jit.compile
def sampling(A: gs.Matrix, t: torch.Tensor):
    return A[:, t * 2], A[t * 1, :], A[t * 1, t * 2], A[:, :]


_graph = torch.classes.gs_classes.Graph(torch.ones(10))
m = gs.Matrix(_graph)
t = torch.ones(10)
for i in sampling(m, t):
    print(i._graph.get())
```

However, it's still not perfect. There are two questions:
1. How to cache gm (IR) and its global_args after compilation?
2. After compilation with caching, some user's args will not works (e.g. Matrix, fan_out[5,10,15]). Because we have cached them in `global_args`. How to solve it?

### CODE REVIEW

**Comment by gpzlx1** [comment]
Try another interface:
```python
import gs
import torch


def sampling(A: gs.Matrix, t: torch.Tensor):
    return A[:, t * 2], A[t * 1, :], A[t * 1, t * 2], A[:, :]


_graph = torch.classes.gs_classes.Graph(torch.ones(10))
m = gs.Matrix(_graph)
t = torch.ones(10)

comipled_func = gs.jit.compile_class(func=sampling, args=(m, t))

for i in comipled_func(m, t):
    print(i._graph.get())
```

**Comment by gpzlx1** [comment]
New demo for torch.fx
```python
from gs import Graph, Matrix
import gs
import torch

A = Graph(False)
indptr = torch.LongTensor([0, 1, 1, 3, 4]).to('cuda:0')
indices = torch.LongTensor([4, 0, 1, 2]).to('cuda:0')
column_ids = torch.LongTensor([2, 3]).to('cuda:0')
A.load_csc(indptr, indices)
subA = A.columnwise_slicing(column_ids)
subA.print()


def sampling(A: Matrix):
    column_ids = torch.LongTensor([2, 3]).to('cuda:0')
    return A[:, column_ids]


m = Matrix(A)
compiled_func = gs.jit.compile(func=sampling, args=(m, ))
print("\nOrigin:")
sampling(m)._graph.print()

print()

print("\nCompiled:")
print(compiled_func(m, ))
compiled_func(m, )._graph.print()
```

File: src/graph.h [hunk 1]
```diff
@@ -21,6 +21,7 @@ class Graph : public torch::CustomClassHolder {
  private:
   bool is_subgraph_;
   std::shared_ptr<CSC> csc_;
+  torch::Tensor test_trace_data_;
```
Comment (czkkkkkk) [inline src/graph.h]: remove this one

File: python/gs/jit/module.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                new_args.append(static_args[offset])
+            else:
+                raise ValueError
+    return tuple(new_args)
+
+
+class compile:
+
+    def __init__(self, func, args):
+        actions = get_actions(args)
```
Comment (czkkkkkk) [inline python/gs/jit/module.py:83]: Could you have some comments to explain what do we do for the arguments and why we need to do it?

File: python/gs/jit/trace.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import math
+import torch
+from torch.fx import Tracer
+from torch.fx import GraphModule
+
+from ..matrix_api import Matrix
+
+
+class gsTracer(Tracer):
```
[the remark this answers is not in the document]
Comment (gpzlx1) [reply python/gs/jit/trace.py]: done

File: src/graph.h (hunk 1 above)
Comment (gpzlx1) [reply src/graph.h]: done

File: python/gs/jit/module.py (hunk 2 above)
Comment (gpzlx1) [reply python/gs/jit/module.py:103]: done.

**Review by czkkkkkk — APPROVED** [review]

**Pull request merged**